src/domain/Benefit.js

```diff
@@ -0,0 +1,146 @@
+import { DAYS, DISCOUNT, EVENTS, NAMES } from '../constants/events.js';
+import { MENUS } from '../constants/menus.js';
+import { OUTPUT_MESSAGE } from '../constants/texts.js';
+
+class Benefit {
+  #visitDay;
+  #benefitSheet;
+
+  constructor(visitDay) {
+    this.#visitDay = visitDay;
+    this.#benefitSheet = new Map([
+      [EVENTS.season.name, 0],
+      [EVENTS.special.name, 0],
+      [EVENTS.weekday.name, 0],
+      [EVENTS.weekend.name, 0],
+      [EVENTS.gift.name, 0],
+    ]);
+  }
+
+  calcaulteTotalBenefit() {
+    let totalBenefit = 0;
+
+    this.#benefitSheet.forEach(discount => {
+      totalBenefit += discount;
+    });
+
+    return totalBenefit;
+  }
+
+  processBenefitDetails() {
+    let benefitDetails = '';
+
+    if (this.hasNoneDiscounts()) {
+      return EVENTS.none;
+    }
+
+    this.#benefitSheet.forEach((discount, eventType) => {
+      if (discount !== 0) {
+        benefitDetails += OUTPUT_MESSAGE.details(eventType, discount);
+      }
+    });
+
+    return benefitDetails;
+  }
+
+  hasNoneDiscounts() {
+    const noneDiscount = [...this.#benefitSheet.values()].every(
+      discount => discount === 0,
+    );
+
+    return noneDiscount;
+  }
+
+  createBenefitSheet(totalAmount, { dessertCount, mainCount }) {
+    const events = Object.keys(EVENTS);
+
+    events.forEach(eventType => {
+      if (totalAmount >= EVENTS[eventType].condition) {
+        const discount = this.calculateDiscount(eventType, {
+          dessertCount,
+          mainCount,
+        });
+        this.#benefitSheet.set(EVENTS[eventType].name, discount);
+      }
+    });
+
+    return this.processBenefitDetails();
+  }
+
+  calculateDiscount(eventType, { dessertCount, mainCount }) {
+    let discount = 0;
+
+    discount -= this.calculateSeasonDiscount(eventType);
+    discount -= this.calculateWeekdayDiscount(eventType) * dessertCount;
+    discount -= this.calculateWeekendDiscount(eventType) * mainCount;
+    discount -= this.calculateSpecialDiscount(eventType);
+    discount -= this.calculateGiftDiscount(eventType);
+
+    return discount;
+  }
+
+  calculateSeasonDiscount(eventType) {
+    let seasonDiscount = 0;
+    const seasonMap = this.#checkSeason();
+
+    if (seasonMap.has(this.#visitDay) && eventType === NAMES.season) {
+      seasonDiscount += seasonMap.get(this.#visitDay);
+    }
+
+    return seasonDiscount;
+  }
+
+  #checkSeason() {
+    const discounts = new Map();
+    const seasonDays = DAYS.season;
+    let amount = DISCOUNT.seasonStart;
+
+    Array.from({ length: seasonDays }, (_, day) => {
+      discounts.set(day + 1, amount);
+      amount += DISCOUNT.seasonAddDay;
+    });
+
+    return discounts;
+  }
+
+  calculateWeekdayDiscount(eventType) {
+    let weekdayDiscount = 0;
+
+    if (!DAYS.weekend.includes(this.#visitDay) && eventType === NAMES.weekday) {
+      weekdayDiscount += DISCOUNT.weekday;
+    }
+    return weekdayDiscount;
```

Comment:
```suggestion
return !DAYS.weekend.includes(this.#visitDay) && eventType === NAMES.weekday
  ? DISCOUNT.weekday
  : 0;
```

와 같이 사용하는 건 어떻게 생각하시나요?

Reply:
삼항 연산자를 사용하지 않으려고 해서 이렇게 처리했어요..!